examples/island_cluster_with_vm_router/main.tf

```diff
@@ -0,0 +1,219 @@
+      subnet_region = var.region
+      purpose       = "REGIONAL_MANAGED_PROXY"
+      role          = "ACTIVE"
+    },
+    {
+      subnet_name           = "${var.region}-psc-snet-${var.random_suffix}"
+      subnet_ip             = var.psc_subnet_cidr
+      subnet_region         = var.region
+      subnet_private_access = "true"
+      purpose               = "PRIVATE_SERVICE_CONNECT"
+    }
+  ]
+
+  secondary_ranges = {
+    "${var.region}-snet-${var.random_suffix}" = [
+      {
+        range_name    = "${var.region}-snet-pods-${var.random_suffix}"
+        ip_cidr_range = var.secondary_ranges["pods"]
+      },
+      {
+        range_name    = "${var.region}-snet-services-${var.random_suffix}"
+        ip_cidr_range = var.secondary_ranges["services"]
+      },
+    ]
+  }
+
+  routes = flatten([
+    [for k, v in var.primary_net_cidrs :
+      {
+        name              = "egress-gke-${k}-${var.random_suffix}"
+        description       = "egress through the router for range ${v}"
+        destination_range = v
+        tags              = "gke-${var.random_suffix}"
+        next_hop_instance = google_compute_instance.vm.self_link
+        priority          = 100
+      }
+    ],
+    [
+      {
+        name              = "default-igw-${var.random_suffix}"
+        description       = "internet through the router"
+        destination_range = "0.0.0.0/0"
+        tags              = "gke-${var.random_suffix}"
+        next_hop_instance = google_compute_instance.vm.self_link
+        priority          = 100
+      }
+    ]
+  ])
+
+  firewall_rules = [
+    {
+      name      = "iap-fw-${var.random_suffix}"
+      direction = "INGRESS"
+      allow = [
+        {
+          protocol = "TCP"
+          ports    = ["22"]
+        }
+      ]
+      ranges = ["35.235.240.0/20"]
+    },
+    {
+      name      = "tcp-primary-fw-${var.random_suffix}"
+      direction = "INGRESS"
+      allow = [
+        {
+          protocol = "TCP"
+        }
+      ]
+      ranges = [
+        var.subnet_cidr,
+        var.secondary_ranges["pods"]
+      ]
+    },
+  ]
+}
+
+module "gke" {
+  source  = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
+  version = "~> 30.0"
+
+  depends_on = [google_compute_instance.vm]
+
+  name                                 = "gke-test-${var.random_suffix}"
+  project_id                           = var.project_id
+  region                               = var.region
+  release_channel                      = "RAPID"
+  zones                                = var.node_locations
+  network                              = module.net.network_name
+  subnetwork                           = "${var.region}-snet-${var.random_suffix}"
+  ip_range_pods                        = "${var.region}-snet-pods-${var.random_suffix}"
+  ip_range_services                    = "${var.region}-snet-services-${var.random_suffix}"
+  http_load_balancing                  = true
```

Comment:
Let's omit module defaults (here and throughtout)
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/f864e8add729b799529f5ede187347ed5cebd5f9/modules/beta-private-cluster/variables.tf#L96

Reply:
Cleaned up module defaults.